packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewerHighlight/PdfViewerHighlight.tsx

```diff
@@ -54,40 +56,73 @@ const PdfViewerHighlight: FC<Props> = ({
 
   const { textDivs } = pdfRenderedText || {};
 
-  const highlightBoxes = useMemo(() => {
+  const highlightShapes = useMemo(() => {
     highlighter?.setTextContentDivs(textDivs);
     return highlights.map(highlight => {
       return highlighter?.getHighlight(highlight);
     });
   }, [highlighter, highlights, textDivs]);
 
+  const highlightDivRef = useRef<HTMLDivElement | null>(null);
+  useScrollIntoActiveHighlight(highlightDivRef, highlightShapes, activeIds);
+
+  return (
+    <div
+      ref={highlightDivRef}
+      className={cx(`${settings.prefix}--document-preview-pdf-viewer-highlight`, className)}
+    >
+      {highlightShapes.map((shape, hlIndex) => {
+        const active = shape?.highlightId ? activeIds?.includes(shape.highlightId) : false;
+        return (
+          <Highlight
+            key={shape?.highlightId || `highlight_${hlIndex}`}
+            className={highlightClassName}
+            activeClassName={activeHighlightClassName}
+            shape={shape}
+            index={hlIndex}
+            scale={scale}
+            active={active}
+          />
+        );
+      })}
+    </div>
+  );
+};
+
+const Highlight: FC<{
+  className?: string;
+  activeClassName?: string;
+  shape?: HighlightShape;
+  index: number;
+  scale: number;
+  active?: boolean;
+}> = ({ className, activeClassName, shape, index, scale, active }) => {
+  function getPositionStyle(bbox: Bbox, scale: number, padding: number = 0) {
+    const [left, top, right, bottom] = bbox;
+    return {
+      left: `${(left - padding) * scale}px`,
+      top: `${(top - padding) * scale}px`,
+      width: `${(right - left + padding) * scale}px`,
+      height: `${(bottom - top + padding) * scale}px`
+    };
+  }
```

Comment:
Move this function outside of the component, since it's standalone. No sense in redefining it each time the component renders.